src/Sarif/Core/FileData.cs

```diff
@@ -25,7 +27,6 @@ public static FileData Create(Uri uri, SarifWriters.LoggingOptions loggingOption
             var fileData = new FileData()
             {
                 MimeType = mimeType,
-                Uri = uri
```

Comment:
>Uri = uri [](start = 16, length = 9)

Why is this removed?

Reply:
Per other comment, this is the wrong data for this member (which should be a relative uri only for nested items)

---
In reply to: [136628150](https://github.com/Microsoft/sarif-sdk/pull/614#discussion_r136628150) [](ancestors = 136628150)

Reply:
Actually, no:
----
5.15.2:
If the file object represents a top-level file, then the uri property may be present. If present, it shall be equal to the name of the property within run.files (§5.12.9) whose value is this file object. If absent, it shall be interpreted as having that same value.

If the file object represents a nested file whose location relative to its parent can be expressed only by means of a path, then the uri property shall be present, and its value shall be a valid relative URI expressing that path.
----
In other words, for a top-level file, you are allowed to repeat the key as the `uri` property, but you don't have to. So, you can certainly justify this as "not needed, saves space". But it's not the case that this is always a "nested" URI.

---
In reply to: [136632876](https://github.com/Microsoft/sarif-sdk/pull/614#discussion_r136632876) [](ancestors = 136632876,136628150)

Reply:
Oops. ok, will resolve

---
In reply to: [136633850](https://github.com/Microsoft/sarif-sdk/pull/614#discussion_r136633850) [](ancestors = 136633850,136632876,136628150)